src/os/bluestore/BlueFS.cc

```diff
@@ -2106,6 +2130,139 @@ int BlueFS::device_migrate_to_new(
   return 0;
 }
 
+int BlueFS::migrate_wal_to_v1() {
+  _init_logger();
+  string wal_dir("db.wal");
+  auto dir_it = nodes.dir_map.find(wal_dir);
+  if (dir_it == nodes.dir_map.end()) {
+    dout(5) << fmt::format("{} No files needed to move", __func__) << dendl;
+    return 0;
+  }
+  
+  DirRef dir = dir_it->second;
+  
+
+  size_t wal_v2_aggregated_size = 0;
+  std::vector<std::pair<string, FileRef>> files_to_migrate;
+  files_to_migrate.reserve(dir->file_map.size());
+
+  // get info about wal files
+  for (const auto& [file_name, file] : dir->file_map) {
+    if(file->is_new_wal()) {
+      wal_v2_aggregated_size += file->fnode.wal_size;
+      files_to_migrate.push_back({ file_name, file });
+    }
+  }
+
+  // check free space
+  uint8_t prefer_bdev = vselector->select_prefer_bdev(vselector->get_hint_by_dir(wal_dir));
+  uint64_t free_space = alloc[prefer_bdev]->get_free();
+  if (free_space < wal_v2_aggregated_size) {
+    dout(1) << fmt::format("{} Error trying to migrate new WAL V2 to V1. Aggregated size of WAL files is {}, free space is {}", __func__, wal_v2_aggregated_size, free_space) << dendl;
```

Comment:
Why _preferred_ device has to have enough space?
If there will not be enough we will fallback to next-order device.